src/trio/testing/_raises_group.py

```diff
@@ -332,61 +313,107 @@ class RaisesGroup(
 
     even though it generally does not care about the order of the exceptions in the group.
     To avoid the above you should specify the first ValueError with a Matcher as well.
-
-    It is also not typechecked perfectly, and that's likely not possible with the current approach. Most common usage should work without issue though.
     """
 
-    # needed for pyright, since BaseExceptionGroup.__new__ takes two arguments
-    if TYPE_CHECKING:
-
-        def __new__(cls, *args: object, **kwargs: object) -> RaisesGroup[E]: ...
-
-    # allow_unwrapped=True requires: singular exception, exception not being
-    # RaisesGroup instance, match is None, check is None
     @overload
     def __init__(
         self,
         exception: type[E] | Matcher[E],
         *,
         allow_unwrapped: Literal[True],
         flatten_subgroups: bool = False,
-        match: None = None,
-        check: None = None,
     ) -> None: ...
 
-    # flatten_subgroups = True also requires no nested RaisesGroup
     @overload
     def __init__(
         self,
         exception: type[E] | Matcher[E],
         *other_exceptions: type[E] | Matcher[E],
-        allow_unwrapped: Literal[False] = False,
         flatten_subgroups: Literal[True],
         match: str | Pattern[str] | None = None,
         check: Callable[[BaseExceptionGroup[E]], bool] | None = None,
     ) -> None: ...
 
+    # simplify the typevars if possible (the following 3 are equivalent but go simpler->complicated)
```

Comment:
Does this work for `RaisesGroup[RaisesGroup[RaisesGroup[Exception]]]`? Don't think there is a test for it.